R/progress.R

```diff
@@ -393,11 +397,26 @@ pb_render <- function(self, private, tokens) {
 
 pb_update <- function(self, private, ratio, tokens) {
   assert_ratio(ratio)
+  stopifnot(!private$finished)
+
   goal <- floor(ratio * private$total)
   self$tick(goal - private$current, tokens)
 }
 
+pb_message <- function(self, private, msg) {
+  assert_character(msg)
+  stopifnot(!private$finished)
+
+  clear_line(private$stream, private$width)
+  cursor_to_start(private$stream)
+  cat(msg, sep = "\n", file = private$stream)
+  if (!private$finished) {
```

Comment:
Can you pls add `&& private$supported` here?

Reply:
Actually, I think we can do a big `if` here. We might not be able to clear the line is `!supported`.

Reply:
So the current code is fine if `supported`, otherwise just `cat` the message and a newline, I think.

Reply:
Done, sorry!